prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -103,12 +104,14 @@ object FhirPathUtils : Logging {
             }
         } catch (e: Exception) {
             // This is due to a bug in at least the extension() function
-            logger.error(
-                "Unknown error while evaluating FHIR expression $expression for condition. " +
-                    "Setting value of condition to false.",
-                e
-            )
-            false
+            val msg =  when (e) {
+                is FHIRLexerException -> "Syntax error on filter expression"
+                is SchemaException -> e.message ?: "Condition error on filter expression"
+                else -> "Unknown error while evaluating FHIR expression $expression for condition. " +
+                    "Setting value of condition to false."
+            }
+            logger.error(msg, e)
+            throw SchemaException(msg)
```

Comment:
This is not setting the final result to false.  Are we still doing that?

Reply:
Since the method is only used in 2 places, I thought it made more sense to have it return the exception if there is one, and then handle that exception appropriately on the calling function